lib/buffer.js

```diff
@@ -1298,16 +1302,17 @@ function atob(input) {
   let equalCharCount = 0;
 
   for (let n = 0; n < input.length; n++) {
-    const index = ArrayPrototypeIndexOf(
-      kForgivingBase64AllowedChars,
-      StringPrototypeCharCodeAt(input, n));
+    const ch = StringPrototypeCharCodeAt(input, n);
+    const val = kForgivingBase64AllowedChars[ch & 0x7f];
 
-    if (index > 4) {
-      // The first 5 elements of `kForgivingBase64AllowedChars` are
-      // ASCII whitespace char codes.
+    if ((ch | val) & ~0x7f) {
```

Comment:
Can you add a comment to what this line is doing?

Reply:
Yeah, will do. It's some non-obvious bitwise magic. It's checking two things: that `ch` is ASCII and is a valid index of `kForgivingBase64AllowedChars`, and that `val` is not `-1` (i.e. `ch` is a valid base64 character).

This line could actually be OR'd onto an accumulator and checked after the loop is complete to remove a branch, but I figure people here would be opposed to it due to the "no optimization of legacy functions" philosophy. 

On that note, I was originally considering making this loop _entirely_ branchless, but I'm guessing that definitely wouldn't be acceptable. It would also probably require a whole series of comments.

Reply:
I understand. I strongly think that we should add comments to the code as well.